RubberduckTests/Grammar/SelectionExtensionsTests.cs

```diff
@@ -0,0 +1,436 @@
+﻿using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using static Rubberduck.Parsing.Grammar.VBAParser;
+using Rubberduck.VBEditor;
+using RubberduckTests.Mocks;
+using Antlr4.Runtime.Misc;
+using System.Collections.Generic;
+
+namespace RubberduckTests.Grammar
+{
+    [TestClass]
+    public class SelectionExtensionsTests
+    {
+        public class CollectorVBAParserBaseVisitor<Result> : VBAParserBaseVisitor<IList<Result>>
+        {
+            private List<Result> defaultResult = new List<Result>();
+
+            protected override IList<Result> DefaultResult => defaultResult;
+            /*
+            protected override IList<Result> AggregateResult(IList<Result> firstResult, IList<Result> secondResult)
+            {
+                if (firstResult != null && secondResult != null)
+                    return firstResult.Concat(secondResult) as IList<Result>;
+
+                if (secondResult == null)
+                    return firstResult;
+
+                return secondResult;
+            }*/
+        }
+
+        public class SubStmtContextElementCollectorVisitor : CollectorVBAParserBaseVisitor<SubStmtContext>
+        {
+            public override IList<SubStmtContext> VisitSubStmt([NotNull] SubStmtContext context)
+            {
+                DefaultResult.Add(context);
```

Comment:
`DefaultResult` is what every return value gets initialized to. You should not add to that! The field is actually readonly; you can only get around that here because it is a collection. 

The way you try to do things, `IList<SubStmtContext>>` is not appropriate. Doing it like this, you should use a field `SubStmtContexts` holding the collection and add to that in a visitor for type `void`.